### PR TITLE
Lunarfusion/css 117 progress circle core tokens

### PR DESCRIPTION
**Pull request by lunarfusion**
## Description
BREAKING CHANGE: This migrates the Progress Circle component to core tokens.
- Jira issue CSS-117 https://jira.corp.adobe.com/browse/CSS-117

## How and where has this been tested?
 - Chrome 103 for macOS
 - Firefox 102 for macOS
 - Safari 15.5 for macOS
 - **How this was tested:** comparing local build of http://localhost:3000/docs/progresscircle.html with XD redline components batch and with https://spectrum.adobe.com/page/progress-circle/

## Screenshots
Screen Shot 2022-08-01 at 11 08 10 AM

Screen Shot 2022-08-01 at 11 08 18 AM

Screen Shot 2022-08-01 at 2 33 04 PM

## To-do list

- [ ] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [x] I have tested these changes in Windows High Contrast mode.

- [ ] This pull request is ready to merge.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,5 +14,7 @@ npm-debug.log*
 /backstop_data/ci_report/
 /backstop_data/backstop_scenarios.json
 /backstop_data/packageDependentMap.json
+backstop_data/engine_scripts/
+backstop.json
```
Comment (yosevu) [inline .gitignore:18]: I don't think we have to ignore these files.

- engine_scripts look like source code
- backstop.json isn't in the root directory

Do you have a backstop.json in your project root?

File: components/progresscircle/package.json [hunk 2]
```diff
@@ -16,15 +16,15 @@
     "build": "gulp"
   },
   "peerDependencies": {
-    "@spectrum-css/vars": "^8.0.0"
+    "@spectrum-css/tokens": "^1.0.1"
```
Comment (yosevu) [inline components/progresscircle/package.json:19]: What was vars again? Does tokens replace it?

**Review by bernhard-adobe — APPROVED** [review]
other than the question about the latest tokens `    "@spectrum-css/tokens": "^1.0.1"` (do `yarn install` please to test it out) this looks really good! Great work.

I think the VRT will fail as there are a couple of class-names renamed. We need to expect that in the test.

File: .gitignore (hunk 1 above)
Comment (lunarfusion) [reply .gitignore:18]: I do, probably from trying to troubleshoot VRT issues. 

@jianliao can I delete backstop.json from my root? Might it have wound up there for a reason while trying to get VRT working? Should I be adding backstop_data/engine_scripts/ to my gitignore, or should that directory be deleted as well?

File: components/progresscircle/package.json (hunk 2 above)
Comment (lunarfusion) [reply components/progresscircle/package.json]: Looks like I had a couple issues getting this file updated. I've corrected it and pushed a [fix commit](https://github.com/adobe/spectrum-css/commit/54c65d61df7931a47fe04bf796a5640ab66ff53b)

File: .gitignore (hunk 1 above)
Comment (pfulton) [reply .gitignore:18]: Yeah, I think we should leave the `.gitignore` as it is, without these two additions. I think you're right @lunarfusion - they might have been due to your trying to troubleshoot VRT issues.

**Comment by pfulton** [comment]
> I think the VRT will fail as there are a couple of class-names renamed. We need to expect that in the test.

@lunarfusion @bernhard-adobe - we actually have Progress Circle ignored in the VRT due to BackstopJS not being able to handle animations, so this one won't even run.

**Review by pfulton — CHANGES_REQUESTED** [review]
@lunarfusion nice work! Let's remove the changes to the `.gitignore` and I think we're good to go on this one.

File: .gitignore (hunk 1 above)
Comment (lunarfusion) [reply .gitignore]: New [commit](https://github.com/adobe/spectrum-css/pull/1488/commits/d8fa0d0ca97daedd4cdbe3465a95c37a85aa9acb) resolves this.

**Review by bernhard-adobe — APPROVED** [review]
Great work, approved

**Comment by bernhard-adobe** [comment]
VRT test: 	https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/35/artifact/backstop_data/html_report/index.html
[2022-08-10 -- 1488 - progress-circle.zip](https://github.com/adobe/spectrum-css/files/9306234/2022-08-10.--.1488.-.progress-circle.zip)

**Review by pfulton — APPROVED** [review]

**Comment by pfulton** [comment]
Released:
`2.0.0-beta.0`

**Comment by pfulton** [comment]
@jnurthen - would you mind taking a look at the changes to the [Progress Circle here](https://github.com/adobe/spectrum-web-components/pull/2538#issuecomment-1262755538) to let @lunarfusion know if the Windows High Contrast Mode styles are acceptable? Thank you!

**Comment by jnurthen** [comment]
@pfulton - I think I made a mistake when I fixed this one originally. The Highlight colour is not guaranteed to be a visible colour combination with the track (which I think gets ButtonBorder/ButtonText by default). 
This is particularly apparent when viewing the High Contrast (light) theme.
Screen Shot 2022-09-29 at 2 17 13 PM


I think a border between the 2 could resolve this - or maybe using HighlightText as one of the colours or another way? However, I wouldn't want to block this PR on that as it is not a new error. IMO This can be tracked in a separate (lower priority) Issue as the visibiity of the progress on a progresscircle is likely low priority.

**Comment by lunarfusion** [comment]
@jnurthen The change that I initially made to WHCM was to ensure the track and the fill have contrast. I did not override the track color for WHCM, but I did override the fill with keyword "Highlight". Here is the VRT result.

**Before:**
188222934-87765339-1058-4c1f-b4b9-29d03f645424

**After:**
188222970-d73f40f3-b829-4858-8576-4d552dfb3503

Note that the VRTs appear to be using Mac emulation of WHCM. Our assessment of the difference between Mac emulation and actual default WHCM colors on a windows machine is:
- Highlight color that looks bright blue (to me) on Mac emulation (left in the visual below)
- Highlight color that looks pale purple (to me) on a PC that a colleague tested for us (right in the visual below)
Screen Shot 2022-09-30 at 2 45 01 PM

Would you like any changes to track or fill color keywords or to the delineation of track and fill based on this info? I can provide a comprehensive visual of our Mac emulation vs PC results if that might help.

**Comment by jnurthen** [comment]
These are the 4 high contrast themes that windows (10) provides. 
### HC #1
Screen Shot 2022-09-30 at 12 23 59 PM
### HC #2
Screen Shot 2022-09-30 at 12 24 14 PM
### HC Dark
Screen Shot 2022-09-30 at 12 23 12 PM
### HC Light
Screen Shot 2022-09-30 at 12 23 38 PM

These were tested using Assistiv Labs which is running windows.

I don't like using forced-color-adjust: none here as that could lead to the track completely disappearing which I don't think is any better than the current situation. 

I was playing around with it and changing the border-style on the track to double might so the trick!

Screen Shot 2022-09-30 at 12 38 05 PM
Screen Shot 2022-09-30 at 12 38 17 PM

**Comment by pfulton** [comment]
Released: `@spectrum-css/progresscircle@2.0.0-beta.1`

**Pull request merged**